### PR TITLE
creating an implementing base components

### PR DESCRIPTION
**Pull request by brett-onions**
## Release Notes

- **New Features**
  - Introduced `BaseDataGrid` for customizable data grid interfaces.
  - Added `BasePageTemplate` for consistent page layouts with titles and optional buttons.

- **Improvements**
  - Enhanced `ClientRoleForm`, `ListRoles`, and `ClientsList` components with new layout structures using `BasePageTemplate` and `BaseDataGrid`.

- **Bug Fixes**
  - Maintained existing functionality and error handling across updated components.

These updates improve usability and maintainability, providing a more cohesive user experience.

### CODE REVIEW

**Comment by rcrichton** [comment]
Task linked: [CU-86c0qwg19 Create base template components](https://app.clickup.com/t/86c0qwg19)

**Review by ItsMurumba — APPROVED** [review]
LGTM!

**Review by marvinIsSacul — APPROVED** [review]

**Comment by drizzentic** [comment]
@coderabbitai review

**Review by ItsMurumba — APPROVED** [review]
LGTM

**Review by drizzentic — APPROVED** [review]
LGTM! Just a small documentation update

File: packages/clients-app/src/pages/clients-list/clients-list.tsx [hunk 1]
```diff
@@ -107,76 +86,34 @@ const ClientsList = () => {
...
+        rows={clients}
+        columns={columns}
+        noRowsOverlay={CustomNoRowsOverlay}
+        onRowClick={params =>
+          window.history.pushState(
+            {},
+            '',
+            '/#!/clients/edit/' + params.row['_id']
+          )
+        }
```
[the remark this answers is not in the document]
Comment (brett-onions) [reply packages/clients-app/src/pages/clients-list/clients-list.tsx:129]: Does not work with hash based routing

File: packages/client-roles-app/src/pages/client-role-form/client-role-form.tsx [hunk 2]
```diff
@@ -156,95 +154,83 @@ export const ClientRoleForm = () => {
...
+        </Box>
+
+        <Divider />
+        <Box sx={{display: 'flex', justifyContent: 'flex-start', p: 1}}>
+          <Button
+            onClick={() => window.history.pushState({}, '', '/#!/client-roles')}
+            variant="outlined"
+          >
+            Cancel
+          </Button>
```
[the remark this answers is not in the document]
Comment (brett-onions) [reply packages/client-roles-app/src/pages/client-role-form/client-role-form.tsx:224]: UseNavigation does not work with hash based routing

**Pull request merged**